tests/models/ecm/test_asoh.py

```diff
@@ -33,3 +38,97 @@ def test_json(full_asoh):
     ):
         assert name_a == name_b
         assert np.allclose(value_a, value_b)
+
+
+def test_energy():
+    # Prepare fields for A-SOH
+    qt = MaxTheoreticalCapacity(base_values=10.)
+    r0 = Resistance(base_values=1.)
+    ocv_ref = ReferenceOCV(base_values=np.ones(11))
+    ocv_ent = EntropicOCV(base_values=np.zeros(11))
+    ocv = OpenCircuitVoltage(ocv_ref=ocv_ref, ocv_ent=ocv_ent)
+    asoh = ECMASOH(q_t=qt, ce=1, ocv=ocv, r0=r0)
+
+    # With this simple example, we expect the energy to simply be the total discharge capacity multiplied by 1 V
+    expected = asoh.q_t.amp_hour
+    computed = asoh.get_theoretical_energy()
+    assert len(computed.shape) == 2, f'Computed energy is not 2D, but has shape {computed.shape}'
+    assert np.allclose(expected, computed), \
+        f'Mismatch in constant OCV ECMASOH energy calculation; expected {expected} but got {computed} Wh!'
```

Comment:
I like these messages from assert statements, but want to make sure you know they're not necessary. In fact, pytest will display the values of `expected` and `computed` in the failure message.

Reply:
Got it, but I don't mind it too much. Makes it easier for me at least, since I run the tests from the terminal like a noob